### PR TITLE
Adds some documentation and fixes error code

### LINKED ISSUE #427: RakuDist test fails on debian
Hi Fernando!

RakuDist test fails on Debian:

https://rakudist.raku.org/sparky/report/debian/333

```
01:47:19 06/19/2020 [bash: zef install Red] ===> Testing: Red:ver<0.1.8>:auth<Fernando Correa de Oliveira>:api<2>
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Failed test 'right exception type (X::Tree::ExistsInTheSameArea)'
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # at t/08-best-tree.t line 29
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Expected: X::Tree::ExistsInTheSameArea
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Got:      X::Red::Driver::Mapped::UnknownError
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Exception message:     Unknown Error!!!
01:49:23 06/19/2020 [bash: zef install Red] [Red]     #     Please, copy this backtrace and open an issue on https://github.com/FCO/Red/issues/new
01:49:23 06/19/2020 [bash: zef install Red] [Red]     #     Driver: Red::Driver::SQLite
01:49:23 06/19/2020 [bash: zef install Red] [Red]     #     Original error: X::DBDish::DBError.new(driver-name => "DBDish::SQLite", native-message => "UNIQUE constraint failed: tree.latitude, tree.longitude", code => 1555, why => "Error")
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # 
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Original error:
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # DBDish::SQLite: Error: UNIQUE constraint failed: tree.latitude, tree.longitude (1555)
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # You failed 1 test of 2
01:49:23 06/19/2020 [bash: zef install Red] [Red] # Failed test 'did we throws-like X::Tree::ExistsInTheSameArea?'
01:49:23 06/19/2020 [bash: zef install Red] [Red] # at t/08-best-tree.t line 51
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Failed test 'right exception type (X::Tree::ExistsInTheSameArea)'
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # at t/08-best-tree.t line 29
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Expected: X::Tree::ExistsInTheSameArea
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Got:      X::Red::Driver::Mapped::UnknownError
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Exception message:     Unknown Error!!!
01:49:23 06/19/2020 [bash: zef install Red] [Red]     #     Please, copy this backtrace and open an issue on https://github.com/FCO/Red/issues/new
01:49:23 06/19/2020 [bash: zef install Red] [Red]     #     Driver: Red::Driver::SQLite
01:49:23 06/19/2020 [bash: zef install Red] [Red]     #     Original error: X::DBDish::DBError.new(driver-name => "DBDish::SQLite", native-message => "UNIQUE constraint failed: tree.latitude, tree.longitude", code => 1555, why => "Error")
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # 
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # Original error:
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # DBDish::SQLite: Error: UNIQUE constraint failed: tree.latitude, tree.longitude (1555)
01:49:23 06/19/2020 [bash: zef install Red] [Red]     # You failed 1 test of 2
01:49:23 06/19/2020 [bash: zef install Red] [Red] # Failed test 'did we throws-like X::Tree::ExistsInTheSameArea?'
01:49:23 06/19/2020 [bash: zef install Red] [Red] # at t/08-best-tree.t line 61
01:49:23 06/19/2020 [bash: zef install Red] [Red] # You failed 2 tests of 15
01:51:05 06/19/2020 [bash: zef install Red] [Red] Use of Nil in numeric context
```

HTH

Aleksei

### LINKED ISSUE #428: Red failing to install with Raku 2020.06
Exactly the same error as #427 
```
===> Testing: Red:ver<0.1.8>:auth<Fernando Correa de Oliveira>:api<2>
[Red]     # Failed test 'right exception type (X::Tree::ExistsInTheSameArea)'
[Red]     # at t/08-best-tree.t line 29
[Red]     # Expected: X::Tree::ExistsInTheSameArea
[Red]     # Got:      X::Red::Driver::Mapped::UnknownError
[Red]     # Exception message:     Unknown Error!!!
[Red]     #     Please, copy this backtrace and open an issue on https://github.com/FCO/Red/issues/new
[Red]     #     Driver: Red::Driver::SQLite
[Red]     #     Original error: X::DBDish::DBError.new(driver-name => "DBDish::SQLite", native-message => "UNIQUE constraint failed: tree.latitude, tree.longitude", code => 1555, why => "Error")
[Red]     # 
[Red]     # Original error:
[Red]     # DBDish::SQLite: Error: UNIQUE constraint failed: tree.latitude, tree.longitude (1555)
[Red]     # You failed 1 test of 2
[Red] # Failed test 'did we throws-like X::Tree::ExistsInTheSameArea?'
[Red] # at t/08-best-tree.t line 51
[Red]     # Failed test 'right exception type (X::Tree::ExistsInTheSameArea)'
[Red]     # at t/08-best-tree.t line 29
[Red]     # Expected: X::Tree::ExistsInTheSameArea
[Red]     # Got:      X::Red::Driver::Mapped::UnknownError
[Red]     # Exception message:     Unknown Error!!!
[Red]     #     Please, copy this backtrace and open an issue on https://github.com/FCO/Red/issues/new
[Red]     #     Driver: Red::Driver::SQLite
[Red]     #     Original error: X::DBDish::DBError.new(driver-name => "DBDish::SQLite", native-message => "UNIQUE constraint failed: tree.latitude, tree.longitude", code => 1555, why => "Error")
[Red]     # 
[Red]     # Original error:
[Red]     # DBDish::SQLite: Error: UNIQUE constraint failed: tree.latitude, tree.longitude (1555)
[Red]     # You failed 1 test of 2
```

### PR DESCRIPTION
**Pull request by JJ**
But mainly closes #427, closes #428 by adding a new error code.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,7 +26,7 @@ use Red:api<2>;
 model Person {...}
 
 model Post is rw {
-    has Int         $.id        is serial;
+    has Int         $.id        is serial; # an id field is mandatory
```
Comment (FCO) [inline README.md:29]: `id` is not mandatory. Only if you want to be able to update

File: README.md [hunk 2]
```diff
@@ -42,7 +42,7 @@ model Post is rw {
 }
 
 model Person is rw {
-    has Int  $.id            is serial;
+    has Int  $.id            is serial; # an id field is mandatory
```
Comment (FCO) [inline README.md:45]: `id` is not mandatory. Only if you want to be able to update

File: lib/Red/Driver/SQLite.pm6 [hunk 3]
```diff
@@ -156,7 +156,7 @@ multi method default-type-for(Red::Column $
 multi method default-type-for($ --> Str:D) is default {"varchar(255)"}
 
 
-multi method map-exception(X::DBDish::DBError $x where { .?code == 19 and .native-message.starts-with: "UNIQUE constraint failed:" }) {
+multi method map-exception(X::DBDish::DBError $x where { (.?code == 19 or .?code == 1555) and .native-message.starts-with: "UNIQUE constraint failed:" }) {
```
Comment (FCO) [inline lib/Red/Driver/SQLite.pm6:159]: Do you know why the error code has changed? Different versions with different errors? Maybe we need a better way to map errors in the future.

**Comment by FCO** [comment]
Thank you very much for your PR! Very appreciated! :)

File: README.md (hunk 1 above)
Comment (JJ) [reply README.md:29]: I've eliminated one and clarified the other. Both have an id field, so it's maybe convenient to say why it's there.

File: lib/Red/Driver/SQLite.pm6 (hunk 3 above)
Comment (JJ) [reply lib/Red/Driver/SQLite.pm6:159]: I really have no idea. I've left the former one, just on case it's still used and not deprecated.

**Comment by JJ** [comment]
El mar., 23 jun. 2020 a las 12:45, Fernando Correa de Oliveira (<
notifications@github.com>) escribió:

> Thank you very much for your PR! Very appreciated! :)
>

My pleasure. Feel free to edit out the comments to the README if you don't
think they're necessary.

**Comment by FCO** [comment]
Thank you very much!

**Comment by JJ** [comment]
My pleasure.

**Pull request merged**